### PR TITLE
Add Entry General: Vim

### LINKED ISSUE #546: [Entry] General: Vim
### Reason/inspiration (optional)

In the "General" topic, write a concept entry about Vim, an open-source terminal-based text editor for Unix-based systems. Be sure to include the following:

- A link to the official Vim website, https://www.vim.org/.
- Code snippets (```bash) that show how to launch Vim.
- A _brief_ section about how to use Vim.
- A screenshot of Vim being used on a terminal.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by paulpetelski**
### Description

Added a new entry about the vim text editors - content/general/concepts/vim/vim.md

Closes #546 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by paulpetelski** [comment]
Hi, sorry I used my other computer to run the yarn commands but I committed the last changes using my other GitHub account that is signed in on that computer. Is this an issue or do I just need to sign the CLA on that account as well?

**Comment by SSwiniarski** [comment]
> Hi, sorry I used my other computer to run the yarn commands but I committed the last changes using my other GitHub account that is signed in on that computer. Is this an issue or do I just need to sign the CLA on that account as well?

@paulpetelski, That should do it.

**Comment by SSwiniarski** [comment]
@paulpetelski, I'll be doing the first review.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@paulpetelski, I reviewed this and left some broad suggestions.

**Comment by paulpetelski** [comment]
@SSwiniarski Thank you for your feedback! I have implemented your suggestions.

**Review by SSwiniarski — APPROVED** [review]
Let's go on to the second review.

**Comment by Dusch4593** [comment]
> Hi, sorry I used my other computer to run the yarn commands but I committed the last changes using my other GitHub account that is signed in on that computer. Is this an issue or do I just need to sign the CLA on that account as well?

Hey @paulpetelski that should not be an issue as long as all account involved with this PR have signed the CLA (which seems to be the case here 😄). There were two things I would like to run by you, though:

* Is the email associated with the @paulpetelski GH account the same email associated with your Codeademy profile? If not, those would have to be synced in order for your profile to be credited on the live entry.
* Did you want to use a profile image besides what seems to be the default one set for @paulpetelski?

**Comment by paulpetelski** [comment]
> > Hi, sorry I used my other computer to run the yarn commands but I committed the last changes using my other GitHub account that is signed in on that computer. Is this an issue or do I just need to sign the CLA on that account as well?
> 
> Hey @paulpetelski that should not be an issue as long as all account involved with this PR have signed the CLA (which seems to be the case here 😄). There were two things I would like to run by you, though:
> 
> * Is the email associated with the @paulpetelski GH account the same email associated with your Codeademy profile? If not, those would have to be synced in order for your profile to be credited on the live entry.
> * Did you want to use a profile image besides what seems to be the default one set for @paulpetelski?

Great! My email address is different, how can I go about syncing it with my Codecademy profile? Also, I am fine with using the default profile image.

**Comment by Dusch4593** [comment]
@paulpetelski Sure thing, happy to help out! 😄 

To change account/email settings on Codecademy to be the same as what's on your GH:
1. Click on your profile image at the top-right and select "Account + Billing"
2. Under the "Account" tab, click on the "Change" next to "Email" to update it.
3. Save the changes.

To sync your Codecademy profile with your GH account (for credit on entries):
1. Log out of your account completely.
2. On the sign-in screen, click on the GitHub icon to sign in through there.
3. You should be good to go! 👍

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:35]:
```suggestion
![Vim screenshot](https://raw.githubusercontent.com/Codecademy/docs/main/media/vim.png)
```

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:12]:
```suggestion
  - 'learn-the-command-line'
  - 'paths/computer-science'
```

The `CatalogContent` should always include two links:

 - One free course slug.
 - One paid `paths/` slug.

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [reply content/general/concepts/vim/vim.md:35]: Also could we switch this image with what the screen would look like after running just `vim`?

Screen Shot 2022-03-17 at 9 54 27 AM

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:28]:
```suggestion

Vim can be run directly from the command line interface or from its GUI.

```

> from its GUI interface

Could we include a small screenshot of this, as well?

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:32]: Maybe a short sentence or two below this snippet explaining what the command is doing? Is it opening a new Vim window with the contents of `filename.txt`?

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@paulpetelski Finished with review 2️⃣ 

I'm pretty excited for this one b/c I've always been curious to know more about Vim. Thank you for this contribution! 🙏 

Tag me whenever you're finished going through and incorporating my comments/suggestions. 😄

**Comment by Dusch4593** [comment]
@paulpetelski Hey there! Thanks for implementing those changes 👍 

> I am having trouble getting the new screenshots to show up using the full file path you suggested. They only show up when I use (/media/vim-terminal.png), I was wondering if you could help me fix this.

I think this is because the entry hasn't yet gone live and,  therefore, the full file path won't work (not yet, at least). It renders locally with `/media/vim-terminal.png` because this is actual relative path reference to that file.

> Also I am getting an error when running yarn format:verify but I do not see what is causing the error. Thank you!

It doesn't look like any of the GH checks are failing (woot! 🙌). If `format:verify` _does_ bring up a file in the future, however, I would just do one of the following:

- Update/save the file(s) and then run `git add` + `commit` and let our [husky hook](https://github.com/Codecademy/docs/blob/main/.husky/pre-commit) perform the lint/format for you.
  - If you go with this option, just know that this only seems to work when you make changes _locally_ (with an editor + terminal) and not anywhere inside GitHub (on the internet).
- After running `format:verify` you could run `format path/to/this/file path/to/that/file` to format multiple files (or run `format:all` to format all of them).

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:3]:
```suggestion
Description: 'Vim is a free, open source text editor that is customizable text and available on many operating systems.'
```

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:44]:
```suggestion
```

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:34]:
```suggestion
Vim can be launch directly from the terminal by running the `vim` command. The start screen will look like this:

![Vim terminal screenshot](https://raw.githubusercontent.com/Codecademy/docs/main/media/vim-terminal.png)
```

File: content/general/concepts/vim/vim.md
Comment (Dusch4593) [inline content/general/concepts/vim/vim.md:42]:
```suggestion
## Vim GUI
```

**Review by Dusch4593 — APPROVED** [review]
@paulpetelski I left a few last-minute comments/suggestions but after that we can go ahead with the next review! 👍

**Comment by paulpetelski** [comment]
> @paulpetelski I left a few last-minute comments/suggestions but after that we can go ahead with the next review! 👍

Thank you! I applied all your suggestions!

**Review by Dusch4593 — APPROVED** [review]
LGTM! 👍 @yangc95 @KTom101 should we conduct one more review for this entry before merging?

File: content/general/concepts/vim/vim.md
Comment (KTom101) [inline content/general/concepts/vim/vim.md:34]:
```suggestion
Vim can be launched directly from the terminal by running the `vim` command. The start screen will look like this:
```

**Review by KTom101 — APPROVED** [review]
Looks good, only saw one minor change

**Comment by paulpetelski** [comment]
@KTom101 Thank you, I have made the correction!

File: content/general/concepts/vim/vim.md
Comment (yangc95) [inline content/general/concepts/vim/vim.md:22]:
```suggestion
This section describes some important features of the Vim editor.
```

File: content/general/concepts/vim/vim.md
Comment (yangc95) [inline content/general/concepts/vim/vim.md:27]: Do you mean "using keyboard shortcuts"?

File: content/general/concepts/vim/vim.md
Comment (yangc95) [inline content/general/concepts/vim/vim.md:44]:
```suggestion
## Vim Graphical User Interface (GUI)
```

**Comment by yangc95** [comment]
@paulpetelski two suggestions! Tys for your work on this entry ☺️

**Comment by paulpetelski** [comment]
@yangc95 Thank you! I implemented your suggestions!

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
4 👀 > 1 👁️ ! Went ahead and merged this after four solid reviews 🚀 

Live link: https://www.codecademy.com/resources/docs/general/vim

TY @Dusch4593 @KTom101 @SSwiniarski

**Pull request merged**
